elyra/pipeline/pipeline_definition.py

```diff
@@ -0,0 +1,281 @@
+
+    @property
+    def id(self) -> str:
+        return self._pipeline_definition.get('id')
+
+    @property
+    def schema_version(self) -> str:
+        return self._pipeline_definition.get('version')
+
+    @property
+    def pipelines(self) -> list:
+        if 'pipelines' not in self._pipeline_definition:
+            raise ValueError("Pipeline is missing 'pipelines' field.")
+        elif len(self._pipeline_definition['pipelines']) == 0:
+            raise ValueError("Pipeline has zero length 'pipelines' field.")
+
+        pipelines: list = list()
+        for pipeline in self._pipeline_definition['pipelines']:
+            pipelines.append(Pipeline(pipeline))
+
+        return pipelines
+
+    @property
+    def primary_pipeline(self) -> Pipeline:
+        if not self._primary_pipeline:
+            if "pipelines" not in self._pipeline_definition:
+                raise ValueError("Pipeline is missing 'pipelines' field.")
+            elif len(self._pipeline_definition['pipelines']) == 0:
+                raise ValueError("Pipeline has zero length 'pipelines' field.")
+
+            # Find primary pipeline
+            self._primary_pipeline = self._get_pipeline_definition(self._pipeline_definition['primary_pipeline'])
+
+            assert self._primary_pipeline is not None, "No primary pipeline was found"
+
+        return Pipeline(self._pipeline_definition, self._primary_pipeline)
+
+    def validate(self) -> list:
+        validation_issues = list()
+        # Validate pipeline schema version
+        if 'version' not in self._pipeline_definition:
+            validation_issues.append("Pipeline schema version field is missing.")
+        elif not isinstance(self._pipeline_definition['version'], str):
+            validation_issues.append("Pipeline schema version field should be a string.")
+
+        # Validate pipelines
+        if 'pipelines' not in self._pipeline_definition:
+            validation_issues.append("Pipeline is missing 'pipelines' field.")
+        elif not isinstance(self._pipeline_definition["pipelines"], list):
+            validation_issues.append("Field 'pipelines' should be a list.")
+        elif len(self._pipeline_definition['pipelines']) == 0:
+            validation_issues.append("Pipeline has zero length 'pipelines' field.")
+
+        # Validate primary pipeline
+        if 'primary_pipeline' not in self._pipeline_definition:
+            validation_issues.append("Could not determine the primary pipeline.")
+        elif not isinstance(self._pipeline_definition["primary_pipeline"], str):
+            validation_issues.append("Field 'primary_pipeline' should be a string.")
+
+        primary_pipeline = self._get_pipeline_definition(self._pipeline_definition.get('primary_pipeline'))
+        if not primary_pipeline:
+            validation_issues.append("No primary pipeline was found")
+        else:
+            # Validate primary pipeline structure
```

Comment:
I think this is where we call `validate(issues_list)` on each pipeline entry.  A pipeline entry should determine if its _the_ primary pipeline in its constructor, then the code below can just be part of the Pipeline.validate() method and conditional if it's the primary pipeline.